newton/_src/sim/builder.py

```
@@ -4605,6 +4605,147 @@ def add_shape_convex_hull(
+        All subsequently added shapes will use SDF collision detection.
+
+        Args:
+            max_resolution: Maximum SDF grid dimension (must be divisible by 8). Common: 32, 64, 128, 256
+            narrow_band_range: (inner, outer) distance range for SDF computation in meters
+            contact_margin: Contact detection margin in meters
+        """
+        if max_resolution % 8 != 0:
+            raise ValueError(
+                f"max_resolution must be divisible by 8 (got {max_resolution}). "
+                "This is required because SDF volumes are allocated in 8x8x8 tiles."
+            )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)